src/content/tutorial/react-step-2/react-step-2.mdx

```diff
@@ -0,0 +1,641 @@
+        attached_groups: 'Andrews VM Groups',
+        status: 'Active',
+      },
+    ];
+    const headers = [
+      {
+        key: 'name',
+        header: 'Name',
+      },
+      {
+        key: 'protocol',
+        header: 'Protocol',
+      },
+      {
+        key: 'port',
+        header: 'Port',
+      },
+      {
+        key: 'rule',
+        header: 'Rule',
+      },
+      {
+        key: 'attached_groups',
+        header: 'Attached Groups',
+      },
+      {
+        key: 'status',
+        header: 'Status',
+      },
+    ];
+    return (
+    ...
+    );
+```
+
+We'll start by adding `<DataTable/>` to our second column. It will have three props: `rows`, which will accept the `initialRows` array we created earlier, `headers`, which will accept the `headers` array we created earlier, and `render`, which will accept a function to render the rest of the data table components.
+
+```javascript
+<div className="bx--col-lg-13">
+  <DataTable rows={initialRows} headers={headers} render={() => ()} />
+</div>
+```
+
+#### Render function
+
+Next we'll add the following arguments to our render function:
+
+```javascript
+render={({
+  rows,
+  headers,
+  getHeaderProps,
+  getRowProps,
+  getTableProps,
+}) => ()}
+```
+
+In the render function, the structure of our data table components will be like so:
+
+```bash
+TableContainer
+└──Table
+   ├──TableHead
+   │  └───TableRow
+   │      ├──TableExpandHeader
+   │      └──TableHeader
+   └──TableBody
+      └───React.Fragment
+          ├──TableExpandRow
+          │  └──TableCell
+          └──TableExpandedRow
+```
+
+In our file, that will look like this:
+
+<!-- prettier-ignore-start -->
+```javascript
+render={({ rows, headers, getHeaderProps, getRowProps, getTableProps, }) => (
+<TableContainer
+  title="Data table title"
+  description="Additional description if needed"
+>
+  <Table {...getTableProps()}>
+    <TableHead>
+      <TableRow>
+        <TableExpandHeader />
+        {headers.map(header => (
+        <TableHeader {...getHeaderProps({ header })}>
+          {header.header}
+        </TableHeader>
+        ))}
+      </TableRow>
+    </TableHead>
+    <TableBody>
+      {rows.map(row => (
+      <React.Fragment key="{row.id}">
+        <TableExpandRow {...getRowProps({ row })}>
+          {row.cells.map(cell => (
+          <TableCell key="{cell.id}">{cell.value}</TableCell>
+          ))}
+        </TableExpandRow>
+        <TableExpandedRow colSpan="{headers.length" + 1}>
+          <h1>Expandable row content</h1>
+          <p>Description here</p>
+        </TableExpandedRow>
+      </React.Fragment>
+      ))}
+    </TableBody>
+  </Table>
+</TableContainer>
+)}
+```
+<!-- prettier-ignore-end -->
+
+#### Pagination
+
+Last but not least, we'll need to add our pagination component as a sibling of `<DataTable/>`.
+
+```javascript
+<DataTable render={({ rows, headers, getHeaderProps, getRowProps, getTableProps,
+}) => ( ... )} />
+<Pagination
+  totalItems="{1228}"
```

Comment:
Shouldn't have quotes

Reply:
Might need to wrap this in Prettier ignore comments. I'm guess Prettier is adding the quotes.